### PR TITLE
gotktrix: init at 0.1.1

### PR DESCRIPTION
**Pull request by chuangzhu**
###### Motivation for this change

Add gotktrix, a Matrix client in Go and GTK4.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/instant-messengers/gotktrix/default.nix [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  };
+  vendorSha256 = "sha256-eO/2MvWOVCeeCsiU2mSwgSEVlSbCXOp8qHyoG0lmk+Q=";
+
+  buildInputs = [
+    gtk4
+    glib
+    graphene
+    gdk-pixbuf
+    gobject-introspection
+    gtk3
```
Comment (NickCao) [inline pkgs/applications/networking/instant-messengers/gotktrix/default.nix:34]: Isn't gtk4 alone enough?

File: pkgs/applications/networking/instant-messengers/gotktrix/default.nix [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    glib
+    gobject-introspection
+  ];
+
+  nativeBuildInputs = [ pkg-config ];
+
+  # Checking requires a working display
+  doCheck = false;
+  meta = with lib; {
+    description = "Matrix client in Go and GTK4";
```
Comment (papojari) [inline pkgs/applications/networking/instant-messengers/gotktrix/default.nix:35]:
```suggestion
    description = "Matrix client written in Go and GTK4";
```

File: pkgs/top-level/all-packages.nix [hunk 3]
```diff
@@ -1882,6 +1882,10 @@ with pkgs;
 
   gopacked = callPackage ../applications/misc/gopacked { };
 
+  gotktrix = callPackage ../applications/networking/instant-messengers/gotktrix {
+    buildGoModule = buildGo117Module;
```
Comment (papojari) [inline pkgs/top-level/all-packages.nix:1886]: Isn't the `buildGoModule` like… unnecessary?

**Comment by papojari** [comment]
Also building gets stuck on `Building subPackage .` [Already opened an issue.](https://github.com/diamondburned/gotktrix/issues/28)

**Review by auroraanna — CHANGES_REQUESTED** [review]

File: pkgs/top-level/all-packages.nix (hunk 3 above)
Comment (chuangzhu) [reply pkgs/top-level/all-packages.nix:1886]: I just realized that `buildGoModule` is now `buildGo117Module`.

**Comment by papojari** [comment]
> Also building gets stuck on `Building subPackage .` [Already opened an issue.](https://github.com/diamondburned/gotktrix/issues/28)

It did build in the end.

**Review by auroraanna — APPROVED** [review]

**Comment by qbit** [comment]
Builds and tests fine for me!

File: pkgs/applications/networking/instant-messengers/gotktrix/default.nix [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  buildInputs = [
+    gtk4
+    glib
+    gobject-introspection
+  ];
+
+  nativeBuildInputs = [ pkg-config ];
+
+  # Checking requires a working display
+  doCheck = false;
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/instant-messengers/gotktrix/default.nix:32]:
```suggestion
  doCheck = false;

```

File: pkgs/applications/networking/instant-messengers/gotktrix/default.nix [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+buildGoModule rec {
+  pname = "gotktrix";
+  version = "0.1.1";
+
+  src = fetchFromGitHub {
+    owner = "diamondburned";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-9feKg/rnkWdJRolHBQ5WT6Rl3xTFe82M8HyxJK3VuN4=";
+  };
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/instant-messengers/gotktrix/default.nix:20]:
```suggestion
  };

```

**Pull request merged**